R/geom-hline.r

```diff
@@ -9,8 +9,26 @@ geom_hline <- function(mapping = NULL, data = NULL,
                        na.rm = FALSE,
                        show.legend = NA) {
 
+  # Warn if supplied mapping is going to be overwritten
+  if (!missing(yintercept) && !missing(mapping)) {
+    warning(paste0("Using both `yintercept` and `mapping` may not have the desired result as mapping is overwritten if `yintercept` is specified\n",
+                   "  Consider placing `yintercept` inside your `aes()` call.\n",
+                   "  e.g. `aes(yintercept=2,colour=colour)`"
+            )
+    )
+  }
+
   # Act like an annotation
   if (!missing(yintercept)) {
+    # Warn if supplied mapping is going to be overwritten
+    if (!missing(yintercept) && !missing(mapping)) {
+      warning(paste0("Using both `yintercept` and `mapping` may not have the desired result as mapping is overwritten if `yintercept` is specified\n",
```

Comment:
You don't need to check `!missing(yintercept)` again (this is the reason that this code belongs in this block).

For now, I think it would be better if the warning message simply stated what was happening (i.e. setting `yintercept` overrides `mapping`) rather than recommending a solution.